syntax = "proto3";

import "common/proto/signature_info.proto";

package resdb;

message QC {
    repeated SignatureInfo signatures = 1;
};

message POERequest {
    enum Type {
      TYPE_NONE = 0;
      TYPE_CLIENT_REQUEST = 1;
      TYPE_NEWREQUEST = 2;
      TYPE_PROPOSE = 3;
      TYPE_SUPPORT = 4;
      TYPE_CERTIFY = 5;
      TYPE_VIEWCHANGE = 6;
    };
  
    int32 type = 1;
    bytes data = 2;
    SignatureInfo data_signature = 3;
    int32 sender_id = 4;
    uint64 view = 5;
    uint64 seq = 6;
    bytes hash = 8;
    QC qc = 9;
    int32 proxy_id = 10; 
}

message CertifyRequests {
	repeated POERequest certify_requests = 1;
}

message ViewChangeMsg {
	CertifyRequests certify_requests = 1;
	uint64 view = 2;
	int32 sender_id = 3;
	SignatureInfo view_signature = 4;
}

message NewViewChangeMsg {
	CertifyRequests certify_requests = 1;
	uint64 view = 2;
	    int32 sender_id = 3;
    QC qc = 4;
}

